addons/controllers/antrea/antreaconfig_controller.go

```diff
@@ -62,6 +62,11 @@ func (r *AntreaConfigReconciler) Reconcile(ctx context.Context, req ctrl.Request
 	// deep copy AntreaConfig to avoid issues if in the future other controllers where interacting with the same copy
 	antreaConfig = antreaConfig.DeepCopy()
 
+	// skip reconciliation for AntreaConfig CR used as template
+	if _, ok := antreaConfig.Annotations[constants.TKGAnnotationTemplateConfig]; ok {
```

Comment:
**thought**: Probably would be better to do this in a predicate function.

e.g.: https://github.com/vmware-tanzu/tanzu-framework/blob/main/addons/controllers/clusterbootstrap_controller.go#L109

Reply:
Thanks for the review! Using predicate makes sense, I've updated this PR to incorporate new [commit](https://github.com/vmware-tanzu/tanzu-framework/pull/2721/commits/d2ac754386d68f4fb36dfe7d35af535533244c4b) which adds predicate.